docs/how-to/rocks/chisel-existing-rock.rst

```diff
@@ -0,0 +1,132 @@
+and thus reducing its attack surface, while retaining its functionality. 
+
+For this, you'll want to ensure that your rock has a ``bare``
+:ref:`base <rockcraft_yaml_base>` and that its contents are
+:ref:`chiselled <chisel_explanation>`.
+
+For this guide, let's take the example of a Python runtime rock.
+
+A Python rock
+-------------
+
+Our starting point is an Ubuntu 22.04-based Python rock, described by the
+following *rockcraft.yaml* file:
+
+.. literalinclude:: ../code/chisel-existing-rock/rock/rockcraft.yaml
+    :language: yaml
+
+This rock can be built by running:
+
+.. literalinclude:: ../code/chisel-existing-rock/task.yaml
+    :language: bash
+    :start-after: [docs:pack-rock]
+    :end-before: [docs:pack-rock-end]
+    :dedent: 2
+
+The resulting rock (*python_3.11_amd64.rock*) will have approximately **42MB**
+and have a functional Python3.11 interpreter. You can verify that by running a
+very simply "Hello, world" Python script in it, with Docker:
+
+.. literalinclude:: ../code/chisel-existing-rock/task.yaml
+    :language: bash
+    :start-after: [docs:run-rock]
+    :end-before: [docs:run-rock-end]
+    :dedent: 2
+
+This rock will also have other utilities like ``bash`` and ``apt`` which come
+from the rock's underlying Ubuntu 22.04 base.
+
+Rebuild the rock with a ``bare`` base
+-------------------------------------
+
+When starting to prepare the rock for production, the main goal is to get rid
+of all the software that is not necessary at runtime, and the first step
+towards achieving that goal is to use a ``bare``. 
+
+In a separate directory, copy the *rockcraft.yaml* from above and replace
+``base: ubuntu@22.04`` with ``base: bare``. With this change, you must also
+remember to tell Rockcraft which Ubuntu release to use for its build
+environment. Do this by adding ``build-base: ubuntu@22.04`` to the
+*rockcraft.yaml* file, which should now look like this:
+
+.. literalinclude:: ../code/chisel-existing-rock/bare-rock/rockcraft.yaml
+    :language: yaml
+
+Pack this rock with the same ``rockcraft`` command as above:
+
+.. literalinclude:: ../code/chisel-existing-rock/task.yaml
+    :language: bash
+    :start-after: [docs:pack-bare-rock]
+    :end-before: [docs:pack-bare-rock-end]
+    :dedent: 2
+
+This new rock (*bare-python_3.11_amd64.rock*) will now have about **28MB** - a
+~33% size reduction - and also have a functional Python3.11 interpreter. 
+Run the same "Hello, world" Python script as before to confirm:
+
+.. literalinclude:: ../code/chisel-existing-rock/task.yaml
+    :language: bash
+    :start-after: [docs:run-bare-rock]
+    :end-before: [docs:run-bare-rock-end]
+    :dedent: 2
+
+
+The question then is: *how is rockcraft able to produce an equally functional
+Python rock with such a drastic reduction in size?* 
+
+And the answer is: *the rock no longer has the Ubuntu base as its first layer,
+and thus no longer has utilities like ``bash`` and ``apt`` (which aren't needed
+anyway)!*
```

Comment:
I might add something along the lines: _for this use case_

Reply:
done